packages/lib-core/src/runtime/PluginStore.ts

```diff
@@ -12,6 +12,16 @@ import { decodeCodeRefs, getPluginModule } from './coderefs';
 import { PluginLoader } from './PluginLoader';
 import type { PluginLoaderOptions } from './PluginLoader';
 
+export type PluginErrorDetails = { message: string; cause?: unknown } & (
+  | {
+      loadError: true;
+    }
+  | {
+      loadError: false;
+      reportedBy: string;
+    }
```

Comment:
I see what we're trying to do here (`reportedBy` is not needed when `loadError` is true as an error occurs loading the plugin). What are you're thoughts on this alternative approach?
```js
export type PluginLifecycle = 'load' | 'reload' | 'run' | 'destroy';

export type PluginErrorDetails = {
  message: string;
  lifecycle: PluginLifecycle;
  cause?: unknown;
  reportedBy?: string;
};
```